doc/_extensions/table_from_rows.py

```diff
@@ -126,29 +110,114 @@ def run(self):
         if rows is not None:
             rows = [r.strip() for r in rows.split(',') if r.strip() != '']
         else:
-            rows = []
+            raise self.severe('No "rows" content provided for "%s" '
+                              % self.name)
+
+        # join all rows for uniqueness, using the order given in
+        # the :rows: option.
+        all_rows = {}
+        all_rows.update(dict.fromkeys(rows))
+        raw = '\n'.join(self._load_header_and_rows(path, header,
+                                                   all_rows.keys()))
+        lines = statemachine.string2lines(raw)
+        self.state_machine.insert_input(lines, path)
+        return []
 
-        if 'sample-yaml-rows' in self.options:
-            source = self.state_machine.input_lines.source(
-                 self.lineno - self.state_machine.input_offset - 1)
-            source_dir = os.path.dirname(os.path.abspath(source))
-            sample_yaml_rows = self._rows_from_sample_yaml(source_dir)
+class TableFromSampleYaml(TableFromRows):
+
+    option_spec = {'header': directives.unchanged}
+
+    @staticmethod
+    def _merge_boards(boards):
+        """Converts pairs of similar board types to one table entry."""
+
+        merge_pairs = [
+            ("nrf9160dk_nrf9160", "nrf9160dk_nrf9160_ns", "nrf9160dk_nrf9160_and_ns"),
+            ("nrf5340dk_nrf5340_cpuapp", "nrf5340dk_nrf5340_cpuapp_ns", "nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns"),
+            ("nrf5340dk_nrf5340_cpuapp", "nrf5340dk_nrf5340_cpunet", "nrf5340dk_nrf5340_cpuapp_and_cpunet"),
+            ("thingy53_nrf5340_cpuapp", "thingy53_nrf5340_cpuapp_ns", "thingy53_nrf5340_cpuapp_and_cpuapp_ns"),
+        ]
```

Comment:
note: to make this more future proof, you could look for `{x}` + `{x}_ns` or `{x}_cpuapp` + `{x}_cpunet` (maybe using regexes). it's not nice to have a dependency on boards at an extension level.